spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe 'posts#index', type: :feature do
+  before(:each) do
+    @user = User.create(
+      name: 'Diana',
+      photo: 'images/user.svg',
+      bio: 'Web Developer',
+      posts_counter: 0
+    )
+
+    @posts = [
+      @post1 = Post.create(author: @user, title: 'User Post 1', text: 'This is User Post 1', likes_counter: 0,
+                           comments_counter: 0),
+      @post2 = Post.create(author: @user, title: 'User Post 2', text: 'This is User Post 2', likes_counter: 0,
+                           comments_counter: 0),
+      @post3 = Post.create(author: @user, title: 'User Post 3', text: 'This is User Post 3', likes_counter: 0,
+                           comments_counter: 0)
+    ]
+
+    visit user_posts_url(user_id: @user.id)
+  end
+
+  describe '#index page' do
+    before(:each) do
+      visit "/users/#{@user.id}/posts"
+    end
+    it 'can see the user profile picture.' do
+      expect(page).to have_css("img[src='#{@user.photo}']")
+    end
+
+    it 'can see the user username.' do
+      expect(page).to have_content(@user.name.to_s)
+    end
+
+    it 'can see the number of posts the user has written.' do
+      expect(page).to have_content(@user.posts_counter.to_s)
+    end
+
+    it 'can see posts title' do
+      # @posts.each do |post|
+      #   expect(page).to have_content(post.title.to_s)
+      # end
+      expect(page).to have_content 'User Post 1'
+    end
+
+    it 'can see some of the body of the post' do
+      # @posts.each do |post|
+      #   expect(page).to have_content(post.text.to_s)
+      # end
```

Comment:
- [ ] Great job working on this milestone. In order to keep out codebase clean and professional, we should always delete all commented lines of code.